newsrevenuehub/rev-engine
# DEV-3847_flake8_versions_match_precommit_upgrade

#### What's this PR do?

1. update pre-commit hooks
2. ensure poetry and pre-commit flake8 versions match
3. flake8 test revengine/settings/*
4. add couple import os (flake8 issues)
5. add ignore "undefined" revengine/settings/* (the reasons files were ignored, but narrower "exception" is better / will catch other issues)
6. enabled talisman checker (lets see if it works in CI...)
7. checked the dockerlinter issue was till issue (it was so remains commented out)
8. briefly checked pylint. it's too much more change for this ticket. and I don't really pylint...
9. [update] add to pre-commit "safety" ignore django version vulnerability [DEV-4117]
10. [update] fixed version of black, flake8, and isort in pyproject.toml to match those in .pre-commit-config.yaml

As part of upgrade a heap of changes were made by

- black in ./apps/
- js prettier in ./spa/

eslint reports many issues. opened separate ticket [DEV-3848] to fix those.

#### Why are we doing this? How does it help us?
Consistency between dev and CI is good.
In some review noticed flake8 error in revengine/settings. pre-commit check will catch now.

#### Are there detailed, specific, step-by-step testing instructions in the associated ticket?
Yes

#### Did this PR make changes to the user interface that may require changes to the user-facing docs?
N/A

#### Have automated unit tests been added? If not, why?
N/A

#### How should this change be communicated to end users?
N/A

#### Are there any smells or added technical debt to note?
Suppressing the many bandit issues of medium and lower severity.

#### Has this been documented? If so, where?
N/A

#### What are the relevant tickets? Add a link to any relevant ones.
[DEV-3847] broke out resolving eslint to [DEV-3848] 

#### Do any changes need to be made before deployment to production 
Should not.

#### Are there next steps? If so, what? Have tickets been opened for them? List next-step tickets here:

eslint reports many issues, resolving them is up to [DEV-3848]

[DEV-3848]: https://news-revenue-hub.atlassian.net/browse/DEV-3848?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DEV-4117]

[DEV-4117]: https://news-revenue-hub.atlassian.net/browse/DEV-4117?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
[DEV-3847]: https://news-revenue-hub.atlassian.net/browse/DEV-3847?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ
